rezerwacje-app/src/main/java/org/niewidoczniakademicy/rezerwacje/core/CSVService.java

```diff
@@ -0,0 +1,48 @@
+package org.niewidoczniakademicy.rezerwacje.core;
+
+import com.opencsv.bean.CsvToBean;
+import com.opencsv.bean.CsvToBeanBuilder;
+import org.niewidoczniakademicy.rezerwacje.core.model.Room;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class CSVService {
+
+    private Logger logger = LoggerFactory.getLogger(CSVService.class);
+
+    private <T> List<T> parseFile(MultipartFile file,
+                                  Class<? extends T> clazz) throws ParseException {
+        try {
+            if (file.isEmpty()) {
+                return new ArrayList<>();
+            }
+
+            try (InputStream inputStream = file.getInputStream();
+                 InputStreamReader reader = new InputStreamReader(inputStream)) {
+
+                CsvToBean<T> toBean = new CsvToBeanBuilder<T>(reader)
+                        .withType(clazz)
+                        .build();
+                return toBean.parse();
+            }
+        }
+        catch (IOException | RuntimeException e) {
+            logger.info("Error parsing CSV: " + e.getLocalizedMessage());
```

Comment:
Loggera będzie można potem uzyć np @Slf4j 
A z wyjątkami to będzie można utworzyć globalny ControllerExceptionHandler ale to potem więc narazie chyba git